sklearn/naive_bayes.py

```diff
@@ -1215,6 +1215,10 @@ def _joint_log_likelihood(self, X):
         jll = np.zeros((X.shape[0], self.class_count_.shape[0]))
         for i in range(self.n_features_):
             indices = X[:, i]
+            num_categories = self.feature_log_prob_[i].shape[1]
+            if np.max(indices) >= num_categories:
+                raise ValueError("A category unseen during training is present"
+                                 " in feature %d" % i)
             jll += self.feature_log_prob_[i][:, indices].T
```

Comment:
using a try-except around this would be a little more efficient (avoids an additional pass over indices).

Reply:
Fair enough, implemented in https://github.com/scikit-learn/scikit-learn/pull/16045/commits/ee92fce2f916534614f8d6c82240e68e42cef8c1